test/unittests/test_utils_test.cpp

```diff
@@ -70,3 +70,109 @@ TEST(test_utils, print_c_execution_result)
+    str_value_i64.str({});
+    str_value_i64 << TypedExecutionResult{ExecutionResult{Value{-3_u64}}, ValType::i64};
+    EXPECT_EQ(str_value_i64.str(), "result(18446744073709551613 [0xfffffffffffffffd] (i64))");
+
+    std::stringstream str_value_f32;
+    str_value_f32 << TypedExecutionResult{ExecutionResult{Value{1.125f}}, ValType::f32};
+    EXPECT_EQ(str_value_f32.str(), "result(1.125 (f32))");
+    str_value_f32.str({});
+    str_value_f32 << TypedExecutionResult{ExecutionResult{Value{-1.125f}}, ValType::f32};
+    EXPECT_EQ(str_value_f32.str(), "result(-1.125 (f32))");
+
+    std::stringstream str_value_f64;
+    str_value_f64 << TypedExecutionResult{ExecutionResult{Value{1.125}}, ValType::f64};
+    EXPECT_EQ(str_value_f64.str(), "result(1.125 (f64))");
+    str_value_f64.str({});
+    str_value_f64 << TypedExecutionResult{ExecutionResult{Value{-1.125}}, ValType::f64};
+    EXPECT_EQ(str_value_f64.str(), "result(-1.125 (f64))");
+}
+
+TEST(test_utils, result_value_matcher)
+{
+    // Exercise every check in Result(value) implementation.
+    // The implementation and checks below are organized by the value's type in Result(value).
+    using testing::Not;
+
+    // TypedExecutionResult is required to be matched against Result(value).
+    EXPECT_THAT(ExecutionResult(Value{1_u64}), Not(Result(1_u64)));
+
+    EXPECT_THAT(TypedExecutionResult(fizzy::Void, {}), Not(Result(0)));
+    EXPECT_THAT(TypedExecutionResult(fizzy::Trap, {}), Not(Result(0)));
+
+    EXPECT_THAT(TypedExecutionResult(Value{0.0f}, ValType::f32), Result(0.0f));
+    EXPECT_THAT(TypedExecutionResult(Value{0.0}, ValType::f64), Not(Result(0.0f)));
+
+    EXPECT_THAT(TypedExecutionResult(Value{0.0}, ValType::f64), Result(0.0));
+    EXPECT_THAT(TypedExecutionResult(Value{0.0f}, ValType::f32), Not(Result(0.0)));
+
+    EXPECT_THAT(TypedExecutionResult(Value{0_u64}, ValType::i64), Result(0_u64));
+    EXPECT_THAT(TypedExecutionResult(Value{0_u32}, ValType::i32), Not(Result(0_u64)));
+
+    EXPECT_THAT(TypedExecutionResult(Value{0_u32}, ValType::i32), Result(0_u32));
+
+    // For non-negative values zero-extension is conveniently allowed.
+    EXPECT_THAT(TypedExecutionResult(Value{0_u64}, ValType::i64), Result(0));
+    EXPECT_THAT(TypedExecutionResult(Value{0_u64}, ValType::i64), Result(0_u32));
+
+    EXPECT_THAT(TypedExecutionResult(Value{-1_u32}, ValType::i32), Result(-1));
+    EXPECT_THAT(TypedExecutionResult(Value{-1_u32}, ValType::i32), Result(-1_u32));
+    EXPECT_THAT(TypedExecutionResult(Value{-1_u32}, ValType::i32), Not(Result(-1_u64)));
+
+    EXPECT_THAT(TypedExecutionResult(Value{-1_u64}, ValType::i64), Result(-1_u64));
+    EXPECT_THAT(TypedExecutionResult(Value{-1_u64}, ValType::i64), Not(Result(-1)));
+    EXPECT_THAT(TypedExecutionResult(Value{-1_u64}, ValType::i64), Not(Result(-1_u32)));
+}
+
+TEST(test_utils, result_value_matcher_explain_missing_result_type)
+{
+    const auto result = testing::internal::MakePredicateFormatterFromMatcher(Result(1_u64))(
```

Comment:
Well this is clearly something internal to google.mock, I'm not sure it's a good idea to use it. For coverage tests with `Not` above should be sufficient.

Reply:
Ah for coverage you mentioned that tests wth `Not` don't print output... Ok fine, we can leave this.

Reply:
This is to test the message output. Using `Not()` will get us coverage (if `IsInterested()` is not used) but we cannot test the message. This was the previous version. I can roll back to that one if preferred.